api/src/main/java/marquez/db/RunDao.java

```diff
@@ -228,10 +183,87 @@ ExtendedRunRow upsert(
       UUID jobVersionUuid,
       UUID runArgsUuid,
       Instant nominalStartTime,
-      Instant nominalEndTime);
+      Instant nominalEndTime,
+      UUID namespaceUuid,
+      String namespaceName,
+      String jobName,
+      String location);
 
   @SqlUpdate(
       "INSERT INTO runs_input_mapping (run_uuid, dataset_version_uuid) "
           + "VALUES (:runUuid, :datasetVersionUuid) ON CONFLICT DO NOTHING")
   void updateInputMapping(UUID runUuid, UUID datasetVersionUuid);
+
+  @Transaction
+  default void notifyJobChange(UUID runUuid, JobRow jobRow, JobMeta jobMeta) {
+    DatasetDao datasetDao = createDatasetDao();
+
+    upsertRun(runUuid, jobRow.getName(), jobRow.getNamespaceName());
+
+    if (jobMeta.getInputs() != null) {
```

Comment:
does this need to deal with outputs too?

Reply:
No, there is a comment on the function above it that communicates why. There is an alternative route to registering /output/ datasets in the dataset api. The run service only registers inputs. I can try to spruce this up so its not so confusing.